internal/multi_thread_gemm.h

```diff
@@ -558,19 +563,7 @@ inline int HowManyThreads(int max_num_threads, int rows, int cols, int depth) {
   }
 
   // Determine the maximum number of threads.
-  int max_count = max_num_threads;
-  // The special value 0 means try to determine the total number of cores.
-  if (max_count == 0) {
-    // No user-set maximum number of threads, so we need to
-    // do some hardware detection.
-    // This is expensive to query so we do it only once.
-    // Too bad for dynamicness. Also, we dont use the c++11 standard getter
-    // because Google's coding style currently bans #include <thread_>.
-    static const int hardware_threads_count =
-        static_cast<int>(sysconf(_SC_NPROCESSORS_CONF));
-
-    max_count = hardware_threads_count;
-  }
+  int max_count = ResolveMaxThreads(max_num_threads);
```

Comment:
Hard to guess what ResolveMaxThreads means. Can we instead have

int GetHardwareConcurrency();

which would be the above sysconf on POSIX and which you'd implement differently on Windows?

This would also help minimizing the part of the code that is specialized for each platform.